### DISCUSSION TITLE
Considering getting a trimlight system

### DISCUSSION THREAD

**Discussion by davebeiler** [ecdye/TrimPy#6] [General]
Could this be integrated into home-assistant.io (it uses python)

Also does this use all local control? IE no cloud based connection required?

**Comment by ecdye** [comment]
> Could this be integrated into home-assistant.io (it uses python)

It could be integrated with any number of Home Automation services. I just call the module while running a rule in openHAB (similar to home assistant) as it accomplishes what I want pretty well. I won't personally be making any integrations like that but the core API provided here is public and anyone can use it to do so.

> Also does this use all local control? IE no cloud based connection required?

It uses all local control, yes. No cloud connection is required or anything like that, as long as it can connect to the network your Trimlight is on it will work.

**Comment by davebeiler** [comment]
Thanks for the quick response! Could you share a screenshot how you do it?

**Comment by ecdye** [comment]
My personal openHAB EMCAScript(JavaScript) script that is run when triggered by my local sundown looks like this:
``` javascript
var Ephemeris = Java.type("org.openhab.core.model.script.actions.Ephemeris");
var Exec = Java.type("org.openhab.core.model.script.actions.Exec");
var logger = Java.type("org.slf4j.LoggerFactory").getLogger("org.openhab.rule." + ctx.ruleUID);
var now = new Date();

var pattern = "DEFAULT";
if (now.getMonth() == 10) {
  pattern = "HALLOWEEN";
} else if (now.getMonth() == 11) {
  pattern = "THANKSGIVING";
  if (Ephemeris.getDaysUntil("THANKSGIVING") > Ephemeris.getDaysUntil("CHRISTMAS")) {
    pattern = "CHRISTMAS";
  }
} else if (now.getMonth() == 12) {
  pattern = "CHRISTMAS";
}
if ((Ephemeris.getDaysUntil("NEW_YEAR") <= 1) || (Ephemeris.getDaysUntil("NEW_YEAR") >= 360)) {
  pattern = "NEW_YEAR";
} else if ((Ephemeris.getDaysUntil("ST_PATRICK", "/etc/openhab/services/events.xml") <= 3) || (Ephemeris.getDaysUntil("ST_PATRICK", "/etc/openhab/services/events.xml") >= 363)) {
  pattern = "ST_PATRICK";
} else if ((Ephemeris.getDaysUntil("MOTHERS_DAY", "/etc/openhab/services/events.xml") <= 3) || (Ephemeris.getDaysUntil("MOTHERS_DAY", "/etc/openhab/services/events.xml") >= 363)) {
  pattern = "MOTHERS_DAY";
} else if ((Ephemeris.getDaysUntil("INDEPENDENCE_DAY") <= 3) || (Ephemeris.getDaysUntil("INDEPENDENCE_DAY") >= 363)) {
  pattern = "INDEPENDENCE_DAY";
} else if ((Ephemeris.getDaysUntil("ST_VALENTINE", "/etc/openhab/services/events.xml") <= 3) || (Ephemeris.getDaysUntil("ST_VALENTINE", "/etc/openhab/services/events.xml") >= 363)) {
  pattern = "ST_VALENTINE";
} else if ((Ephemeris.getDaysUntil("christian.EASTER", "/etc/openhab/services/events.xml") <= 3) || (Ephemeris.getDaysUntil("christian.EASTER", "/etc/openhab/services/events.xml") >= 363)) {
  pattern = "10";  // Easter does not have a built-in name
}

logger.info("Trimlight changed to mode: manual, and pattern: " + pattern);
Exec.executeCommandLine("python3", "-m", "TrimPy", "--ip", "123.456.789.0", "--mode", "manual", "--pattern", pattern);
```

If you use openHAB yours could look very similar, but I'm not sure what an HA rule to do this would look like.

**Comment by xgalaxy** [comment]
This is awesome. Thank you so much. I’m curious about some things.

I’m on the Trimlight Edge app. What sort of compatibility is there between Select and Edge? Does anyone know?
In my version of the app (1.11.43):
-  I have up to 9 custom color slots and 12 predefined colors (one of which is "off")
-  There are a few more animation options
-  [An "effect" editor](https://www.youtube.com/watch?v=ij6LWLsquTE) that I’m not sure what it does exactly (still new to the app) but some of the options have similar names to the animation options.

**Reply by ecdye** [reply]
I don't personally use the Edge system and my understanding is that it uses a completely different API, so as of now, no I do not believe that this will work with Trimlight Edge. You are welcome to experiment and try and get it working with Edge and I would be happy to work with you on including that code here.

Because of the lack of any compatibility and no formal documentation from the Trimlight folks I don't think that there will be any sort of inter-compatible app from them so trying to get this to work on Edge without a lot of reverse engineering and packet sniffing will be difficult as that was the process used to develop this script.

I hope this answers your question, let me know if you don't understand anything or would like further clarification.

**Comment by spkane** [comment]
For the most record, the most recently released Trimlight Edge API is documented via a link that can be found here:

https://trimlight.com/support/does-the-edge-have-an-open-api

Direct Link (as of this comment): https://39877180.fs1.hubspotusercontent-na1.net/hubfs/39877180/Manuals/Trimlight_Edge_API_Documentation%208192022.pdf

**Reply by ecdye** [reply]
This is cool, it's a little outside the scope of what I would do with this project, but is a great reference for anyone who comes here looking for support for Trimlight Edge
